tests/test_transform.py

```diff
@@ -180,10 +296,126 @@ def checkTransform(self, fromName, toName):
+                self.assertEqual(frameSet.getFrame(4).getIdent(), "currFrame")
+
+                fromEndpoint = transform.getFromEndpoint()
+                toEndpoint = transform.getToEndpoint()
+                polyMap = makePolyMap(nIn, nOut)
+
+                # forward transformation of one point
+                rawInPoint = makeRawPointData(nIn)
+                inPoint = fromEndpoint.pointFromData(rawInPoint)
+                outPoint = transform.tranForward(inPoint)
+                rawOutPoint = toEndpoint.dataFromPoint(outPoint)
+                # TODO replace all use of np.allclose with assertFloatsAlmostEqual once DM-9707 is fixed
+                assert_allclose(rawOutPoint, polyMap.tranForward(rawInPoint), err_msg=msg)
+                assert_allclose(rawOutPoint, frameSet.tranForward(rawInPoint), err_msg=msg)
+
+                # inverse transformation of one point;
+                # remember that the inverse will not give the original values!
+                inversePoint = transform.tranInverse(outPoint)
+                rawInversePoint = fromEndpoint.dataFromPoint(inversePoint)
+                assert_allclose(rawInversePoint, polyMap.tranInverse(rawOutPoint), err_msg=msg)
+                assert_allclose(rawInversePoint, frameSet.tranInverse(rawOutPoint), err_msg=msg)
+
+                # forward transformation of an array of points
+                nPoints = 7  # arbitrary
+                rawInArray = makeRawArrayData(nPoints, nIn)
+                inArray = fromEndpoint.arrayFromData(rawInArray)
+                outArray = transform.tranForward(inArray)
+                rawOutArray = toEndpoint.dataFromArray(outArray)
+                self.assertFloatsAlmostEqual(rawOutArray, polyMap.tranForward(rawInArray), msg=msg)
+                self.assertFloatsAlmostEqual(rawOutArray, frameSet.tranForward(rawInArray), msg=msg)
+
+                # inverse transformation of an array of points;
+                # remember that the inverse will not give the original values!
+                inverseArray = transform.tranInverse(outArray)
+                rawInverseArray = fromEndpoint.dataFromArray(inverseArray)
+                self.assertFloatsAlmostEqual(rawInverseArray, polyMap.tranInverse(rawOutArray), msg=msg)
+                self.assertFloatsAlmostEqual(rawInverseArray, frameSet.tranInverse(rawOutArray), msg=msg)
+
+                # If frameSet has a SkyFrame as base or current, try permuting that (in place);
+                # the Transform constructor should undo the permutation.
+                # Warning: you must permute the axes of the SkYFrame once it is in the frameSet,
+                # rather than before adding it to the FrameSet, else the behavior is wrong
+                for permFrameSet, isBasePermuted, isCurrPermuted in permFrameSetIter(frameSet):
+                    if isBasePermuted:
+                        self.assertEqual(permFrameSet.getFrame(astshim.FrameSet.BASE).getLonAxis(), 2)
```

Comment:
The important part of the test is that after adding a `SkyFrame` to a `Transform` that its axes are *not* permuted, and to be sure this is true I must verify that the axes actually were permuted.

So I don't think the case you are describing is very important. It is also difficult to test because one can only call `getLonAxis` on a `SkyFrame`, so an extra conditional would be required.